File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        if len(s) == 0:
            return 0
        left, right = 0, 0
        longest = 1
        chars = set()
        while right < len(s):
            while s[right] in chars:
                chars.remove(s[left])
                left += 1
            chars.add(s[right])
            longest = max(longest, right - left + 1)
            right += 1
        return longest
        

        

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        //Sliding window
        if (s.size() == 0){
            return 0;
        }
        int left, right = 0;
        int longest = 1;
        unordered_set<char> chars;
        while (right < s.size()){
            while (chars.find(s[right]) != chars.end()){
                chars.erase(s[left]);
                ++left;
            }
            chars.insert(s[right]);
            longest = max(longest, right - left + 1);
            ++right;
        }
        return longest;
    }
};